test/registration_spec.rb

```diff
@@ -201,38 +201,48 @@
 
   describe "#get_updates_list" do
     let(:base_product) { { "name" => "base" } }
+    let(:installer_update_base_product) { { "name" => "self_update_id" } }
     let(:remote_product) { { "name" => "base" } }
     let(:updates) { ["http://updates.suse.com/sles12/"] }
     let(:suse_connect) { double("suse_connect") }
+    let(:self_update_id) { "self_update_id" }
 
     before do
       allow(Registration::SwMgmt).to receive(:base_product_to_register).and_return(base_product)
+      allow(Yast::ProductFeatures).to receive(:GetStringFeature)
+        .with("globals", "self_update_id")
+        .and_return(self_update_id)
```

Comment:
Here is where the self_update_id variable is used, so "self_update_id" for defined and "" for not defined

Reply:
ah ok